tests/test_surface/test_file_io.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+import deprecation
+
+from hypothesis import given, settings
+from hypothesis import strategies as st
+import numpy as np
+
+from xtgeo import RegularSurface, Cube
+import xtgeo
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+@pytest.mark.parametrize("engine", ["cxtgeo", "python"])
+@pytest.mark.parametrize(
+    "fformat",
+    [
+        "irap_binary",
+        "irap_ascii",
+        "zmap_ascii",
+        # "ijxyz",  # This fails horribly
+        "petromod",
+        "zmap",
+        "xtgregsurf",
+    ],
+)
+@pytest.mark.parametrize(
+    "input_val, expected_result",
+    [
+        (1, [[1.0, 1.0], [1.0, 1.0]]),
+        ([1, 2, 3, 4], [[1.0, 2.0], [3.0, 4.0]]),
+        (np.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.ones((2, 2)), [[1.0, 1.0], [1.0, 1.0]]),
+    ],
+)
+def test_simple_io(input_val, expected_result, fformat, engine):
+    if engine == "python" and fformat not in ["irap_ascii", "irap_binary"]:
+        pytest.skip("Only one engine available")
+    if engine == "cxtgeo" and fformat == "irap_binary":
+        pytest.xfail("Fails for unknown reason")
+    surf = RegularSurface(2, 2, 0.0, 0.0, values=input_val)
+    surf.to_file("my_file", fformat=fformat)
+    surf_from_file = RegularSurface.read_file("my_file", fformat=fformat, engine=engine)
+    assert surf_from_file.values.data.tolist() == expected_result
+
+
+@settings(deadline=None)
+@pytest.mark.usefixtures("setup_tmpdir")
+@pytest.mark.parametrize("engine", ["cxtgeo", "python"])
+@pytest.mark.parametrize(
+    "fformat",
+    [
+        "irap_binary",
+        "irap_ascii",
+        "zmap_ascii",
+        # "ijxyz",  # This fails horribly
+        "petromod",
+        "zmap",
+        "xtgregsurf",
+    ],
+)
+@given(
+    data=st.lists(
+        st.floats(
+            allow_nan=False, max_value=xtgeo.UNDEF_LIMIT, min_value=-xtgeo.UNDEF_LIMIT
+        ),
+        min_size=9,
+        max_size=9,
+    )
+)
+def test_complex_io(data, fformat, engine):
```

Comment:
Do we want to vary the size of the grid as well? The orientation? flip etc..?

Reply:
Note that [`@composite`](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.composite) allows for combining strategies, thus one could make a `composite` named e.g. `regular_surface_input` and use it for all the tests creating a regular surface

Reply:
I tried this, and it is possible, but because we are dealing with IO it becomes quite slow (I would have to increase the max runtime for hypothesis), would you still like to see it included? An example of run time is 2 minutes with `nrow` and `ncol` with a 2000 max size.

Reply:
Implemented now, I added max 10 for columns and rows, it did uncover some bugs.